### ISSUE TITLE
Wrong parameter getting assigned in query using createQueryBuilder.

### ISSUE THREAD

**Issue by meAnurag** [typeorm/typeorm#7988]
## Issue Description

So I am trying to build a query using .createQueryBuilder() follwoing is my code:

```typescript

const queryRunner: QueryRunner = getConnection().createQueryRunner();

  await queryRunner.startTransaction();

  try {
    //other queries here deleted for simplicity
    await queryRunner.manager
      .getRepository(User)
      .createQueryBuilder()
      .update(User)
      .set({ status: UserStatus.INACTIVE }) // UserStatus is enum
      .where("organizationId IN (:...ids)", { ids: organizations }) // organizations is number array
      .printSql()
      .execute();

    await queryRunner.commitTransaction();
  } catch (err) {
    await queryRunner.rollbackTransaction();
    console.log("=====Rollback=====");
  } finally {
    await queryRunner.release();
  }

```

### my enviroment:

typeorm: 0.2.25
node : 12.0.0
typescript:  3.8.3
posgresql: 13.3
pg: 8.3.2

### Expected Behavior

when I run the above code the proper output should be a query like:

```
UPDATE "users" SET "status" = $1 WHERE "organizationId" IN ($2, $3, $4) -- PARAMETERS: ["Inactive",35,33,34]
```

### Actual Behavior

but what I get is :

```
UPDATE "users" SET "status" = $2 WHERE "organizationId" IN ($2, $3, $4) -- PARAMETERS: ["Inactive",35,33,34]
```

instead of assigning the first parameter second is being assigned.

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Windows 11  |
| Node.js version     | 12.0.0  | 
| Typescript version  | 3.9.7 | 
| TypeORM version     | 0.2.25  | 

### Relevant Database Driver(s)

| DB Type              | Reproducible  |
| ---                  | ---           |
| `aurora-data-api`    | no            |
| `aurora-data-api-pg` | no            |
| `better-sqlite3`     | no            |
| `cockroachdb`        | no            |
| `cordova`            | no            |
| `expo`               | no            |
| `mongodb`            | no            |
| `mysql`              | no            |
| `nativescript`       | no            |
| `oracle`             | no            |
| `postgres`           | yes           |
| `react-native`       | no            |
| `sap`                | no            |
| `sqlite`             | no            |
| `sqlite-abstract`    | no            |
| `sqljs`              | no            |
| `sqlserver`          | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by blazmrak** [comment]
Have you tried updating to the latest version? I was not able to recreate the bug on my side.

**Comment by imnotjames** [comment]
Yeah - this is a bug that was fixed in the latest release - 0.2.35.

I cannot replicate your issue at this time.
